jwst/stpipe/utilities.py

```diff
@@ -169,9 +171,17 @@ def record_step_status(datamodel, cal_step, success=True):
     else:
         status = SKIPPED
 
+    if isinstance(datamodel, (str, PurePath)):
+        datamodel = datamodels.open(datamodel)
```

Comment:
```suggestion
```
Following up on https://github.com/spacetelescope/jwst/pull/8683#discussion_r1722017139 I don't see how this can support filenames/paths as input. Currently for a filename (of a stored datamodel) it will:
- open the model
- modify the in-memory model
- return without saving

I don't see how this function could re-save the model (as the saving machinery is already very complicated). I suggest instead that we don't add support for filenames/paths and instead update outlier detection to open the file.

Reply:
see comment above and let me know if the next push looks ok